sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -239,5 +247,777 @@ public virtual Response<TimeSeriesModelSettings> UpdateModelSettingsDefaultTypeI
+        /// </remarks>
+        /// <example>
+        /// <code snippet="Snippet:TimeSeriesInsightsSampleGetInstancesByIds">
+        /// </code>
+        /// </example>
+        public virtual async Task<Response<InstancesOperationResult[]>> GetInstancesAsync(
+            IEnumerable<ITimeSeriesId> timeSeriesIds,
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetInstances)}");
+            scope.Start();
+
+            try
+            {
+                Argument.AssertNotNullOrEmpty(timeSeriesIds, nameof(timeSeriesIds));
+
+                InstancesBatchRequest batchRequest = new InstancesBatchRequest()
+                {
+                    Get = new InstancesRequestBatchGetOrDelete()
+                };
+
+                foreach (ITimeSeriesId timeSeriesId in timeSeriesIds)
+                {
+                    batchRequest.Get.TimeSeriesIds.Add(timeSeriesId);
+                }
+
+                Response<InstancesBatchResponse> executeBatchResponse = await _timeSeriesInstancesRestClient
+                    .ExecuteBatchAsync(batchRequest, _clientSessionId, cancellationToken)
+                    .ConfigureAwait(false);
+
+                return Response.FromValue(executeBatchResponse.Value.Get.ToArray(), executeBatchResponse.GetRawResponse());
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets Time Series instances from the environment by Time Series Id.
+        /// </summary>
+        /// <param name="timeSeriesIds">List of Ids of the Time Series instance to return.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>
+        /// List of instance or error objects corresponding by position to the array in the request. Instance object is set when operation is successful
+        /// and error object is set when operation is unsuccessful.
+        /// </returns>
+        /// <seealso cref="GetInstancesAsync(IEnumerable{ITimeSeriesId}, CancellationToken)">
+        /// See the asynchronous version of this method for examples.
+        /// </seealso>
+        public virtual Response<InstancesOperationResult[]> GetInstances(
+            IEnumerable<ITimeSeriesId> timeSeriesIds,
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetInstances)}");
+            scope.Start();
+
+            try
+            {
+                Argument.AssertNotNullOrEmpty(timeSeriesIds, nameof(timeSeriesIds));
```

Comment:
In the case where null is a valid value and let's say customer is using just a single value time series Id, what happens?

Reply:
this will assert that the timeSeriesIds is not null or empty. If a user passes in an array with a null value, both conditions are false.